pytest_nunit/nunit.py

```diff
@@ -171,8 +173,16 @@ def test_cases(self, module):
                 environment=self.environment,
                 settings=None,  # TODO : Add settings as optional fixture
                 failure=FailureType(
-                    message=CdataComment(text=case["error"]),
-                    stack_trace=CdataComment(text=case["stack-trace"]),
+                    message=CdataComment(
```

Comment:
Please can you refactor this so that it only does the `isinstance()` once, e.g.

```python
message = CdataComment(text=str(case["error"]))
if isinstance(case["error"], ExceptionChainRepr):
    stack_trace = CdataComment(text=str(case["error"].reprcrash))
else:
    stack_trace = CdataComment(text=case["stack-trace"])
```

This eliminates the first conditional since `str()` of a string just returns the same string, so they equate to the same thing.

Reply:
Ah if it's a list comprehension then no, that refactor won't work.